assets/js/components/notifications/dashboard-auth-alert.js

```diff
@@ -24,26 +29,95 @@ import { __, sprintf } from '@wordpress/i18n';
+
+	// Selector for `getConnectURL` not merged yet.
+	// const connectURL = useSelect( ( select ) => select( CORE_USER ).getConnectURL() );
+	// TODO: replace use of global with above selector once merged.
+	const connectURL = global.googlesitekit?.admin?.connectURL;
+
+	if ( unsatisfiedScopes === undefined || connectURL === undefined ) {
+		return null;
+	}
+
+	let messageID;
+	let moduleNames;
+	// Determine if all scopes are in Google API format, otherwise use generic message.
+	if ( unsatisfiedScopes.some( ( scope ) => ! scope.match( /^https:\/\/www.googleapis.com\/auth\// ) ) ) {
+		messageID = MESSAGE_GENERIC;
+	} else {
+		// All scopes are in Google API format, map them to module names.
+		moduleNames = mapScopesToModuleNames( unsatisfiedScopes );
+		// If any scope did not resolve to a module name, use the generic message.
+		if ( moduleNames.some( ( name ) => name === false ) ) {
+			messageID = MESSAGE_GENERIC;
+		} else {
+			moduleNames = unique( moduleNames );
+			messageID = 1 < moduleNames.length ? MESSAGE_MULTIPLE : MESSAGE_SINGULAR;
+		}
+	}
+
+	let message;
+
+	switch ( messageID ) {
+		case MESSAGE_MULTIPLE:
+			/* translators: used between list items, there is a space after the comma. */
+			const listSeparator = __( ', ', 'google-site-kit' );
+			message = sprintf(
+				/* translators: %s: List of product names */
+				__( 'Site Kit can’t access the some relevant data because you haven’t granted all permissions requested during setup. To use Site Kit, you’ll need to redo the setup for: %s – make sure to approve all permissions at the authentication stage.', 'google-site-kit' ),
+				moduleNames.join( listSeparator )
+			);
+			break;
+		case MESSAGE_SINGULAR:
+			message = sprintf(
+				/* translators: %1$s: Product name */
+				__( 'Site Kit can’t access the relevant data from %1$s because you haven’t granted all permissions requested during setup. To use Site Kit, you’ll need to redo the setup for %1$s – make sure to approve all permissions at the authentication stage.', 'google-site-kit' ),
+				moduleNames[ 0 ]
+			);
+			break;
+		case MESSAGE_GENERIC:
+			message = __( 'Site Kit can’t access all relevant data because you haven’t granted all permissions requested during setup. Please redo setup.', 'google-site-kit' );
+			break;
+	}
 
 	return (
 		<Notification
 			id="authentication error"
 			title={ __( 'Site Kit can’t access necessary data', 'google-site-kit' ) }
-			/* translators: %1$s: Product name */
-			description={ sprintf( __( 'Site Kit can’t access the relevant data from %1$s because you haven’t granted all API scopes requested during setup. To use Site Kit, you’ll need to redo the setup for %1$s – make sure to approve all API scopes at the authentication stage.', 'google-site-kit' ), product ) }
-			handleDismiss={ () => {} }
+			description={ message }
 			format="small"
 			type="win-error"
-			isDismissable={ true }
+			isDismissable={ false }
```

Comment:
Was this broken before? I see `handleDismiss` above is empty. Otherwise I'm curious why we make the notice non-dismissible here, since that's not part of the ACs afaict?

Reply:
I believe so. I ran into this while testing where the notification should have been displayed but it wasn't because it was dismissed (very confusing). `handleDismiss` is not a prop for this component, but an instance method, so that wouldn't have done anything before.